WD-Firmware/psp/src/psp_util.S

```diff
@@ -0,0 +1,122 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://www.FreeRTOS.org
+ * http://aws.amazon.com/freertos
+ *
+ * 1 tab == 4 spaces!
+ */
+
+/*
+ * The FreeRTOS kernel's RISC-V port is split between the the code that is
+ * common across all currently supported RISC-V chips (implementations of the
+ * RISC-V ISA), and code which tailors the port to a specific RISC-V chip:
+ *
+ * + The code that is common to all RISC-V chips is implemented in
+ *   FreeRTOS\Source\portable\GCC\RISC-V-RV32\portASM.S.  There is only one
+ *   portASM.S file because the same file is used no matter which RISC-V chip is
+ *   in use.
+ *
+ * + The code that tailors the kernel's RISC-V port to a specific RISC-V
+ *   chip is implemented in freertos_risc_v_chip_specific_extensions.h.  There
+ *   is one freertos_risc_v_chip_specific_extensions.h that can be used with any
+ *   RISC-V chip that both includes a standard CLINT and does not add to the
+ *   base set of RISC-V registers.  There are additional
+ *   freertos_risc_v_chip_specific_extensions.h files for RISC-V implementations
+ *   that do not include a standard CLINT or do add to the base set of RISC-V
+ *   registers.
+ *
+ * CARE MUST BE TAKEN TO INCLDUE THE CORRECT
+ * freertos_risc_v_chip_specific_extensions.h HEADER FILE FOR THE CHIP
+ * IN USE.  To include the correct freertos_risc_v_chip_specific_extensions.h
+ * header file ensure the path to the correct header file is in the assembler's
+ * include path.
+ *
+ * This freertos_risc_v_chip_specific_extensions.h is for use on RISC-V chips
+ * that include a standard CLINT and do not add to the base set of RISC-V
+ * registers.
+ *
+ */
+
+#include "psp_specific_definitions.h"
+#include "psp_specific_macros.h"
+#include "psp_asm_macros.h"
+#include "psp_specific_rtos.h"
+#include "rtosal_asm_macros.h"
+
+.global pspHandleEcall
+.global pspInitializeStack
+.extern vTaskSwitchContext
+
+.global pxCurrentTCB
+
+.align 8
+.func
+pspInitializeStack:
+
+	csrr t0, mstatus					/* Obtain current mstatus value. */
+    M_ADDI t1, x0, 0x188                /* Generate the value 0x1880, which are -   */
+	slli t1, t1, 4                      /* the MPIE and MPP bits to set in mstatus. */
+	or t0, t0, t1						/* Set MPIE and MPP bits in mstatus value. */
+
+    M_ADDI a0, a0, -REGBYTES
+    M_STORE t0, 0(a0)                         /* mstatus onto the stack. */
+    M_ADDI a0, a0, -(22 * REGBYTES)           /* Space for registers x11-x31. */

```

Comment:
define instead of '22'. find it in all places

Reply:
Not needed anymore as the code is moved to portASM.S which is in the portable part